src/storage/table/update_segment.cpp

```diff
@@ -695,14 +695,21 @@ string_t UpdateSelectElement::Operation(UpdateSegment &segment, string_t element
 }
 
 template <class T>
-static void InitializeUpdateData(UpdateInfo &base_info, Vector &base_data, UpdateInfo &update_info, Vector &update,
-                                 const SelectionVector &sel) {
-	auto update_data = FlatVector::GetData<T>(update);
+static void InitializeUpdateData(UpdateInfo &base_info, Vector &base_data, UpdateInfo &update_info,
+                                 UnifiedVectorFormat &update, const SelectionVector &sel) {
+	auto update_data = update.GetData<T>(update);
 	auto tuple_data = update_info.GetData<T>();
 
-	for (idx_t i = 0; i < update_info.N; i++) {
-		auto idx = sel.get_index(i);
-		tuple_data[i] = update_data[idx];
+	if (sel.IsSet()) {
```

Comment:
I am not sure about this, this was necessary, because we return `nullptr` from some of the `statistics_update_function` implementations, and that breaks with a ConstantVector.

I wonder if there is a better way to do this, perhaps merging the selection vectors?
And if not, I wonder if there are other places where this should be replicated that we aren't doing currently.

Reply:
But I'll have a go at making these changes

Reply:
In retrospect, you're right
this wasn't the right way to implement this.

The unified vector format's selvec should only be applied when accessing the `update` vector (or its validity mask)